src/System.Net.Http/tests/StressTests/HttpStress/Program.cs

```diff
@@ -610,34 +614,55 @@ int Combine(int h1, int h2)
         GC.KeepAlive(listener);
     }
 
-    private static (string, string) GetGetQueryParameters(string contentSource, RequestContext clientContext, int numParameters)
+    private static string GetGetQueryParameters(ref string uri, int maxRequestLineSize, string contentSource, RequestContext clientContext, int numParameters)
     {
-        StringBuilder queryString = new StringBuilder();
-        StringBuilder expectedString = new StringBuilder();
-        queryString.Append($"?Var{0}={contentSource}");
-        expectedString.Append(contentSource);
+        if (maxRequestLineSize < uri.Length)
+        {
+            throw new ArgumentOutOfRangeException(nameof(maxRequestLineSize));
+        }
+        if (numParameters <= 0)
+        {
+            throw new ArgumentOutOfRangeException(nameof(numParameters));
+        }
+
+        var expectedString = new StringBuilder();
+        var uriSb = new StringBuilder(uri);
+        maxRequestLineSize -= uri.Length;
 
-        int num = clientContext.GetRandomInt(numParameters);
+        int appxMaxValueLength = Math.Max(maxRequestLineSize / numParameters, 1);
 
-        for (int i = 1; i < num; i++)
+        int num = clientContext.GetRandomInt32(1, numParameters + 1);
+        for (int i = 0; i < num; i++)
         {
-            string vari = clientContext.GetRandomSubstring(contentSource);
-            expectedString.Append(vari);
-            queryString.Append($"&Var{i}={vari}");
+            string key = $"{(i == 0 ? "?" : "&")}Var{i}=";
+            if (maxRequestLineSize <= uriSb.Length + key.Length)
+            {
+                break;
+            }
+
+            int remainingLength = maxRequestLineSize - uriSb.Length - key.Length;
+            if (remainingLength <= 0)
+            {
+                break;
+            }
```

Comment:
This check is same as the check previous check
```c#
remaininglength <= 0
maxRequestLineSize - uriSb.Length - key.Length  <= 0
maxRequestLineSize <= uriSb.Length + key.Length
```

we should remove one of these.

Reply:
Yes, this was meant to replace the previous, I just neglected to delete it apparently.